Code/Tools/AssetProcessor/native/utilities/ApplicationManagerBase.cpp

```diff
@@ -1555,8 +1555,37 @@ void ApplicationManagerBase::RegisterBuilderInformation(const AssetBuilderSDK::A
                     retryCount++;
                     result = builderRef->RunJob<AssetBuilder::CreateJobsNetRequest, AssetBuilder::CreateJobsNetResponse>(
                         request, response, s_MaximumCreateJobsTimeSeconds, "create", "", nullptr);
-                } while (result == AssetProcessor::BuilderRunJobOutcome::LostConnection &&
-                         retryCount <= AssetProcessor::RetriesForJobNetworkError);
+
+                    // If a lost connection occured, prepare for a retry using an exponential backoff policy
+                    if ((result == AssetProcessor::BuilderRunJobOutcome::LostConnection || 
+                         result == AssetProcessor::BuilderRunJobOutcome::ProcessTerminated ) && (retryCount <= AssetProcessor::RetriesForJobLostConnection))
+                    {
+                        const int delay = 1 << (retryCount-1);
+
+                        // Check if we need a new builder, and if so, request a new one
+                        if (!builderRef->IsValid())
+                        {
+                            AZ_TracePrintf(AssetProcessor::ConsoleChannel, "Lost connection to builder %s. Retrying (Attempt %d with %d second delay)",
+                                            builderRef->GetUuid().ToString<AZStd::string>().c_str(),
+                                            retryCount,
+                                            delay);
+
+                            builderRef.release();
+                            AssetProcessor::BuilderManagerBus::BroadcastResult(builderRef, &AssetProcessor::BuilderManagerBusTraits::GetBuilder, false);
+                        } 
+                        else
+                        {
+                            AZ_TracePrintf(AssetProcessor::ConsoleChannel, "Lost connection to builder %s. Retrying with a new Builder (Attempt %d  with %d second delay)",
```

Comment:
Should these messages be switched?  This one says its retrying with a new builder but it doesn't actually do anything.  The one above doesn't actually mention a new builder but it does seem to start a new one.

Reply:
Oh you're right, good catch.